saleor/graphql/page/mutations.py

```diff
@@ -83,3 +93,160 @@ class Meta:
+        """
+        if attributes:
+            not_valid_attributes = [
+                graphene.Node.to_global_id("Attribute", attr.pk)
+                for attr in attributes
+                if attr.type != AttributeType.PAGE_TYPE
+            ]
+            if not_valid_attributes:
+                error = ValidationError(
+                    "Only page type attributes allowed.",
+                    code=PageErrorCode.INVALID.value,
+                    params={"attributes": not_valid_attributes},
+                )
+                errors[field].append(error)
+
+
+class PageTypeCreate(PageTypeMixin, ModelMutation):
+    class Arguments:
+        input = PageTypeCreateInput(
+            description="Fields required to create page type.", required=True
+        )
+
+    class Meta:
+        description = "Create a new page type."
+        model = models.PageType
+        permissions = (PageTypePermissions.MANAGE_PAGE_TYPES_AND_ATTRIBUTES,)
+        error_type_class = PageError
+        error_type_field = "page_errors"
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+        errors = defaultdict(list)
+        try:
+            cleaned_input = validate_slug_and_generate_if_needed(
+                instance, "name", cleaned_input
+            )
+        except ValidationError as error:
+            error.code = PageErrorCode.REQUIRED.value
+            errors["slug"].append(error)
+
+        cls.validate_attributes(
+            errors, cleaned_input.get("add_attributes"), "add_attributes"
+        )
+
+        if errors:
+            raise ValidationError(errors)
+
+        return cleaned_input
+
+    @classmethod
+    def _save_m2m(cls, info, instance, cleaned_data):
+        super()._save_m2m(info, instance, cleaned_data)
+        attributes = cleaned_data.get("add_attributes")
+        if attributes is not None:
+            instance.page_attributes.add(*attributes)
+
+
+class PageTypeUpdate(PageTypeMixin, ModelMutation):
+    class Arguments:
+        id = graphene.ID(description="ID of the page type to update.")
+        input = PageTypeUpdateInput(
+            description="Fields required to update page type.", required=True
+        )
+
+    class Meta:
+        description = "Update page type."
+        model = models.PageType
+        permissions = (PageTypePermissions.MANAGE_PAGE_TYPES_AND_ATTRIBUTES,)
+        error_type_class = PageError
+        error_type_field = "page_errors"
+
+    @classmethod
+    def check_for_duplicates(cls, errors: Dict[str, List[ValidationError]], data: dict):
+        """Check if any items are on both list for adding and removing.
+
+        Raise error if some of items are duplicated.
+        """
+        add_attributes = data.get("add_attributes")
+        remove_attributes = data.get("remove_attributes")
+
+        duplicated_ids = get_duplicates_ids(add_attributes, remove_attributes)
```

Comment:
Should we add validation for duplicated ids in one of the lists? 
For example:
remove_attributes = [<id1>, <id2>, <id1>] ?

Reply:
I am not sure if it's necessary. If you enter the same attribute twice it still will be added just once the same situation is with removing.